### PR TITLE
feat: added MetricDetail page

### PR DESCRIPTION
**Pull request by hussedev**

### CODE REVIEW

File: public/avatarPlaceholder.svg
[the remark this answers is not in the document]
Comment (carlbarrdahl) [reply public/avatarPlaceholder.svg:1]: Or maybe easier to just edit the svg to remove the OP Sunny paths.

File: src/features/metrics/components/MetricsSidebar/AllocationList.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        sortAscending ? a.amount - b.amount : b.amount - a.amount,
+      ),
+    [projects, sortAscending],
+  );
+
+  return (
+    <>
+      {projectsSortedByAmount.map((project) => (
+        <AllocationItem key={project.id} {...project}>
+          {handleFormatAllocation(project.amount)}
```
Comment (carlbarrdahl) [inline src/features/metrics/components/MetricsSidebar/AllocationList.tsx:26]: Something with the calculation here. Try using `project.fraction` instead.

image

File: src/components/Text.tsx
[the remark this answers is not in the document]
Comment (hussedev) [reply src/components/Text.tsx:1]: done

File: src/features/metrics/components/MetricsSidebar/SortButton.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const text = useMemo(
+    () => (sortAscending ? "Ascending" : "Descending"),
+    [sortAscending],
+  );
+
+  return (
+    <button
+      onClick={onClick}
+      className="
+        ring-offset-background
```
[the remark this answers is not in the document]
Comment (hussedev) [reply src/features/metrics/components/MetricsSidebar/SortButton.tsx]: I created one, for now I leave it there, I don't wanna invest time now in using the give one as I like how is it now, later we can refactor to use the given Button component.

File: src/components/ScrollArea.tsx
[the remark this answers is not in the document]
Comment (hussedev) [reply src/components/ScrollArea.tsx:1]: done!

File: public/avatarPlaceholder.svg
[the remark this answers is not in the document]
Comment (hussedev) [reply public/avatarPlaceholder.svg:1]: done!

File: src/pages/[domain]/metrics/[metricId]/index.tsx [hunk 3]
```diff
@@ -1,13 +1,30 @@
...
-      <p>Metric: {metricId}</p>
-      <pre>{JSON.stringify(projects, null, 2)}</pre>
-    </div>
+    <MetricsLayout>
+      <div className="mx-auto mb-24 flex max-w-screen-lg gap-8 px-4 pb-32 pt-16">
+        {isPending ? (
+          <div> Loading...</div>
+        ) : (
+          <>
+            <section className="flex-col">
```
[the remark this answers is not in the document]
Comment (hussedev) [reply src/pages/[domain]/metrics/[metricId]/index.tsx]: done

**Pull request merged**